impeller/entity/geometry/stroke_path_geometry.cc

```diff
@@ -574,27 +557,43 @@ GeometryResult StrokePathGeometry::GetPositionBuffer(
     return {};
   }
 
-  Scalar min_size =
-      (pass.GetSampleCount() == SampleCount::kCount4 ? kMinStrokeSizeMSAA
-                                                     : kMinStrokeSize) /
-      max_basis;
+  Scalar min_size = kMinStrokeSize / max_basis;
   Scalar stroke_width = std::max(stroke_width_, min_size);
-
+  bool is_hairline = stroke_width_ <= min_size;
   auto& host_buffer = renderer.GetTransientsBuffer();
-  auto scale = entity.GetTransform().GetMaxBasisLength();
+
+  // This is a harline stroke and can be drawn directly with line primitives,
+  // which avoids extra tessellation work, cap/joins, and overdraw prevention.
+  if (is_hairline && path_.IsSingleContour()) {
```

Comment:
Does IsSingleContour guarantee no overdraw?

Reply:
Why would it need to?

Reply:
You'd need this test to be `IsSingleNonSelfIntersectingContour()`

Reply:
Right, I still need to use the prevent overdraw setting

Reply:
Done